[DataGrid] Fix accessibility issues in panels and toolbar buttons

romgrk opened the pull request:
Fixes https://github.com/mui/mui-x/pull/5796

Based on #5806

## Changes:

- [x] set `aria-controls` to buttons only when the  menu / panel is open
- [x] connect buttons & panels/menu via aria attributes
- [x] clear/reset aria attributes when menu / panel is not open

MBilalShafi commented on packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx:
```diff
@@ -129,8 +137,12 @@ const GridToolbarFilterButton = React.forwardRef<HTMLButtonElement, GridToolbarF
+          id={filterButtonId}
           size="small"
           aria-label={apiRef.current.getLocaleText('toolbarFiltersLabel')}
+          aria-controls={isOpen ? filterPanelId : undefined}
```
Should the filter header button have same `aria` attributes as it seems to drive the same filter panel?
https://github.com/mui/mui-x/blob/efd46f46d1ec9dca880b8366984539ad1f08ec37/packages/grid/x-data-grid/src/components/columnHeaders/GridColumnHeaderFilterIconButton.tsx#L66-L75

MBilalShafi reviewed (approved):
Nice improvements accessibility-wise. Thank you 👍 

A couple small suggestions:

```diff
diff --git a/packages/grid/x-data-grid/src/components/cell/GridActionsCell.tsx b/packages/grid/x-data-grid/src/components/cell/GridActionsCell.tsx
index 5eb94358b..ec2d48da4 100644
--- a/packages/grid/x-data-grid/src/components/cell/GridActionsCell.tsx
+++ b/packages/grid/x-data-grid/src/components/cell/GridActionsCell.tsx
@@ -200,8 +200,8 @@ function GridActionsCell(props: GridActionsCellProps) {
           id={buttonId}
           aria-label={apiRef.current.getLocaleText('actionsCellMore')}
           aria-controls={menuId}
-          aria-expanded={open ? 'true' : undefined}
-          aria-haspopup="true"
+          aria-expanded={open}
+          aria-haspopup
           role="menuitem"
           size="small"
           onClick={showMenu}
diff --git a/packages/grid/x-data-grid/src/components/columnHeaders/ColumnHeaderMenuIcon.tsx b/packages/grid/x-data-grid/src/components/columnHeaders/ColumnHeaderMenuIcon.tsx
index 1eb387b8e..2d119e398 100644
--- a/packages/grid/x-data-grid/src/components/columnHeaders/ColumnHeaderMenuIcon.tsx
+++ b/packages/grid/x-data-grid/src/components/columnHeaders/ColumnHeaderMenuIcon.tsx
@@ -59,9 +59,9 @@ export const ColumnHeaderMenuIcon = React.memo((props: ColumnHeaderMenuIconProps
           aria-label={apiRef.current.getLocaleText('columnMenuLabel')}
           size="small"
           onClick={handleMenuIconClick}
-          aria-expanded={open ? 'true' : undefined}
-          aria-haspopup="true"
-          aria-controls={columnMenuId}
+          aria-expanded={open}
+          aria-haspopup
+          aria-controls={open ? columnMenuId : undefined}
           id={columnMenuButtonId}
           {...rootProps.slotProps?.baseIconButton}
         >
```

romgrk commented:
I've addressed the comments.

@MBilalShafi I've added an explicit `aria-haspopup="menu"` everywhere, it keeps it consistent with other locations in the codebase that also use `aria-haspopup="menu"`. Both `"menu"` and `"true"` are equivalent: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup
I didn't like `aria-haspopup` because the type of the aria attribute is a string, not a boolean.
